### DISCUSSION TITLE
OAuth Vue App

### DISCUSSION THREAD

**Discussion by xi-osman** [directus/directus#7590] [Archive]
Im creating a Vue application that uses Directus <3 as a backend service. I'm trying to authenticate via facebook SSO which I have setup already on Directus but I don't want to use the Admin app I want to use my own Vue app. What is the guide for this? I know I have to hit this endpoint in Directus GET /auth/oauth/facebook but that takes me to a JSON page if I enter that url manually. I don't think I can do an AJAX request since I need OAuth to hop around. 

![image](https://user-images.githubusercontent.com/463779/130724538-2d046955-54e2-4b46-b731-50e67226eed2.png)

**Comment by xi-osman** [comment]
@rijkvanzanten Sorry to pull you in haha any idea on how to implement OAuth on a web app?

**Comment by xi-osman** [comment]
Never mind I think I found out how this is done. Seems like I will need my app and backend (Directus) to be under the same domain. The steps would be the following. 

1. User opens your app at app.example.com
2. They click on the provider they need example facebook 
3. Browser navigates to backend.example.com/auth/oauth/facebook?redirect=app.example.com/oauth
4. Directus + Grant does its magic ❤️
5. Browser gets routed to app.example.com/oauth
6. Now the domain *.example.com should have a new cookie named directus_refresh_token
7. Use the token to do the refresh and bingo you are logged in

**Reply by rijkvanzanten** [reply]
Glad you figured it out!

**Reply by biapar** [reply]
I'm interested. Could you post some details?

> Never mind I think I found out how this is done. Seems like I will need my app and backend (Directus) to be under the same domain. The steps would be the following.
> 
> 1. User opens your app at app.example.com
> 2. They click on the provider they need example facebook
> 3. Browser navigates to backend.example.com/auth/oauth/facebook?redirect=app.example.com/oauth
> 4. Directus + Grant does its magic ❤️
> 5. Browser gets routed to app.example.com/oauth
> 6. Now the domain *.example.com should have a new cookie named directus_refresh_token
> 7. Use the token to do the refresh and bingo you are logged in

I'm interested. Could you post some details?

**Comment by NilsBaumgartner1994** [comment]
@xi-osman can you provide a link to your repo ?
I am building also an web/native app with directus.

**Reply by xi-osman** [reply]
Hey I don't have a public repo but not sure how this would work on a native app I only created a browser app. I was able to authenticate using a sub domain cookie.
